coral/src/app/features/connectors/request/ConnectorPromotionRequest.tsx

```diff
@@ -0,0 +1,364 @@
+  getConnectorDetailsPerEnv,
+  requestConnectorPromotion,
+} from "src/domain/connector";
+import { HTTPError } from "src/services/api";
+import { parseErrorMsg } from "src/services/mutation-utils";
+import { ConnectorSkeletonForm } from "src/app/features/connectors/request/components/ConnectorSkeletonForm";
+import { Dialog } from "src/app/components/Dialog";
+import isEmpty from "lodash/isEmpty";
+
+function ConnectorPromotionRequest() {
+  const [targetEnv, setTargetEnv] = useState<Environment | undefined>(
+    undefined
+  );
+
+  const [sourceEnv, setSourceEnv] = useState<Environment | undefined>(
+    undefined
+  );
+
+  const [cancelDialogVisible, setCancelDialogVisible] = useState(false);
+
+  const { connectorName } = useParams();
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+  const toast = useToast();
+
+  const sourceEnvId = searchParams.get("sourceEnv");
+  const targetEnvId = searchParams.get("targetEnv");
+
+  function showErrorToast(message: string) {
+    toast({
+      message: message,
+      position: "bottom-left",
+      variant: "danger",
+    });
+  }
+
+  // redirect user in case source or target env id are not set
+  useEffect(() => {
+    if (!targetEnvId || !sourceEnvId) {
+      showErrorToast(`Missing url parameter`);
+      navigate(`/connector/${connectorName}`, { replace: true });
+      return;
+    }
+  }, [sourceEnvId, targetEnvId]);
+
+  const {
+    data: environments,
+    isFetched: isFetchedEnvironments,
+    isError: isErrorEnvironments,
+    error: errorEnvironments,
+  } = useQuery<Environment[], Error>({
+    queryKey: ["getAllEnvironmentsForConnector"],
+    queryFn: () => getAllEnvironmentsForConnector(),
+  });
+
+  useEffect(() => {
+    if (isFetchedEnvironments) {
+      // redirect user back in case there is an error
+      if (isErrorEnvironments) {
+        showErrorToast(
+          `Error while fetching available environments: ${parseErrorMsg(
+            errorEnvironments
+          )}`
+        );
+        navigate(`/connector/${connectorName}`, { replace: true });
+        return;
+      }
+
+      // redirect user back in case source or target env do not exist
+      // e.g. because user came from old url
+      const sourceEnvFromUrl = environments?.find((env) => {
+        return env.id === sourceEnvId;
+      });
+
+      const targetEnvFromUrl = environments?.find((env) => {
+        return env.id === targetEnvId;
+      });
+
+      if (!sourceEnvFromUrl) {
+        showErrorToast(
+          `No source environment was found with ID ${sourceEnvId}`
+        );
+        navigate(`/connector/${connectorName}`, { replace: true });
+        return;
+      }
+
+      if (!targetEnvFromUrl) {
+        showErrorToast(
+          `No target environment was found with ID ${targetEnvId}`
+        );
+        navigate(`/connector/${connectorName}`, { replace: true });
+        return;
+      }
+
+      setTargetEnv(targetEnvFromUrl);
```

Comment:
If I understand correctly, we use this state instead of the URL params as the `targetEnv` and `sourceEnv` values we pass as values in the form. 

I think this may be unnecessary? If the `targetEnv` and/or `sourceEnv` are invalid, this `useEffect` will catch it. If not, then the values in the URL params should be correct, and we may rely on them? This would save us some state management.

Reply:
The values from the URL are only the `id`s but we also need the `name` values (so the whole `Environment` object). That's why they are set in case both values from the URL are valid and exist.

Reply:
Ah yes of course! I am blind haha. 

I have another question actually: the dependencies of this useEffect are not exhaustive: https://github.com/Aiven-Open/klaw/pull/1723/files#diff-8f5f615b6d9ff249e567c21197fe62817a85a2f4c430bc26e5774cc95cddb868R132

Usually I'm not extremely into defending this rule, which is often making life harder (for good reasons, but still). However, here, we have flow control dependening on variables which are not dependencies (`isErrorEnvironments`), so it probably should be added ^^

Reply:
You're absolutely right! It's odd, because usually linting (or my IDE?) should show a warning for that. Will add it!